File: src/main/java/com/jacuum/engine/Sessions.java
package com.jacuum.engine;

import com.jacuum.map.GameMap;

public interface Sessions {
    String open(GameMap map, String algoName, String username,
                String avatar, int iterations) throws Exception;
    void start(String id) throws Exception;
    void pause(String id) throws Exception;
    void resume(String id) throws Exception;
    void stop(String id) throws Exception;
    SessionView view(String id) throws Exception;
}

File: src/main/kotlin/com/jacuum/engine/Sessions.kt
package com.jacuum.engine

import com.jacuum.map.GameMap

interface Sessions {
    @Throws(Exception::class)
    fun open(map: GameMap, algoName: String, username: String, avatar: String, iterations: Int): String
    @Throws(Exception::class) fun start(id: String)
    @Throws(Exception::class) fun pause(id: String)
    @Throws(Exception::class) fun resume(id: String)
    @Throws(Exception::class) fun stop(id: String)
    @Throws(Exception::class) fun view(id: String): SessionView
}
